### PR TITLE
Dont use dumb lists

### PR DESCRIPTION
**Pull request by cpdough**
trying to remove the ordinal list matching stuff since it is confusing. dicts make it easier and better for the UI i think. unless @phillip-rasgo  says i'm wrong.

did i miss any transforms that use ordinal list matching?

### CODE REVIEW

File: column_transforms/datepart/datepart.sql [hunk 1]
```diff
@@ -1,3 +1,9 @@
 SELECT *,
 {% for col in date_columns %}DATE_PART('{{date_part}}', {{col}}) as {{col}}_datepart_{{date_part}} {{ ", " if not loop.last else "" }}{% endfor %}
+from {{ source_table }}
+
+SELECT *,
+{%- for target_col, date_part in dates.items() %}
+    DATE({{target_col}}, '{{date_part}}') as {{target_col}}_date {{ ", " if not loop.last else "" }}
+{%- endfor %}
```
Comment (szaghikian-rasgo) [inline column_transforms/datepart/datepart.sql:8]: wait why are there two select statements here? Or are you planning to delete the first in favor of the second?

Comment (szaghikian-rasgo) [reply column_transforms/datepart/datepart.sql:8]: Actually I think that you may have dropped a date cast into this template?

File: column_transforms/new_cast/new_cast.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: new_cast
+description: |
+  Cast selected columns to a new type
+arguments:
+  casts:
+    type: cast_dict
```
Comment (szaghikian-rasgo) [inline column_transforms/new_cast/new_cast.yaml:6]: do we need a new type here? if we're gonna start using k-v more thoroughly in UDTs, should we just have a dict type?

File: column_transforms/datepart/datepart.sql (hunk 1 above)
Comment (cpdough) [reply column_transforms/datepart/datepart.sql:4]: thanks for the find

File: column_transforms/new_cast/new_cast.yaml (hunk 2 above)
Comment (cpdough) [reply column_transforms/new_cast/new_cast.yaml]: good call i forgot about the new column_value_dict just for these types of things

File: column_transforms/datepart/datepart.yaml [hunk 3]
```diff
@@ -1,19 +1,20 @@
...
+  An exhaustive list of valid date parts can be [found here](https://docs.snowflake.com/en/sql-reference/functions-date-time.html#label-supported-date-time-parts).
 arguments:
-  date_columns:
-    type: column_list
-    description: names of column(s) you want to date-part
-  date_part:
-    type: date_part
-    description: the desired grain of the date; an exhaustive list of valid date parts can be [found here](https://docs.snowflake.com/en/sql-reference/functions-date-time.html#label-supported-date-time-parts).
+  dates:
+    type: datepart_dict
```
Comment (phillip-rasgo) [inline column_transforms/datepart/datepart.yaml:8]: This type `datepart_dict` is not yet supported on the UI side.   Should we add it?

File: docs/column_transforms/new_cast.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+# new_cast
+
+Cast selected columns to a new type
+
+
+## Parameters
+
+| Argument |   Type    |                                    Description                                     |
+| -------- | --------- | ---------------------------------------------------------------------------------- |
+| casts    | cast_dict | A dict where the keys are columns and the values are the new type to cast them to. |
```
Comment (phillip-rasgo) [inline docs/column_transforms/new_cast.md:12]: I'm not seeing the corresponding YAML for this, but this look like a new type as well.  (I do think both of these new types are a good idea to add.)

**Pull request merged**